### ISSUE TITLE
Unable to move disk image location to another drive

### ISSUE THREAD

**Issue by Jonx0r** [docker/for-win#14163]
### Description

In Settings > Resources, if I try to change the Disk Image Location from C:\Users\Jonx0r\AppData\Local\Docker\wsl to a location on another drive like D:\Docker\ I get the following error:

<img width="418" alt="Docker_2_ErrorMigratingWSLDisk" src="https://github.com/docker/for-win/assets/110224264/292bda8d-bb1b-4aec-8cbb-4a76f1c8817d">

Disk Image Location reverts to default when I close and relaunch Docker Desktop.

I just need to move the docker_data.vhdx and ext4.vhdx files to a different drive since they balloon like crazy and I'm running out of space on C. 

### Reproduce

1. Launch Docker Desktop.
2. Go to Settings > Resources.
3. Change Disk Image Location to a path on a different drive.

### Expected behavior

For the Disk Image Location to change to another drive successfully. 

### docker version

```bash
Client:
 Version:           26.1.4
 API version:       1.45
 Go version:        go1.21.11
 Git commit:        5650f9b
 Built:             Wed Jun  5 11:29:54 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.31.1 (153621)
 Engine:
  Version:          26.1.4
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.11
  Git commit:       de5c9cf
  Built:            Wed Jun  5 11:29:22 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.33
  GitCommit:        d2d58213f83a351ca8f528a95fbd145f5654e957
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    26.1.4
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.14.1-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.1-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.32
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.24
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.2.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.9.3
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 26.1.4
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d2d58213f83a351ca8f528a95fbd145f5654e957
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.153.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 15.55GiB
 Name: docker-desktop
 ID: 7f9a0aa3-5c47-4ccf-9cec-b9e4ff70b241
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=npipe://\\.\pipe\docker_cli
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

61559C7D-44A3-4D0C-B78B-2C766E8BEF56/20240629232502

**Comment by Jonx0r** [comment]
I should add, this is a fresh install. Haven't used Docker before.

**Comment by Jonx0r** [comment]
Workaround:

1. Close Docker Desktop.
2. Open %APPDATA%\Docker\settings.json.
3. Change the path for 'customWslDistroDir'.
4. Relaunch Docker Desktop.
5. Confirm docker_data.vhdx gets moved to the specified directory.

**Comment by andrea-reale** [comment]
Hi,

thanks for reporting this issue. I believe this is a duplicate of #13384 

We're aware of the problem that Docker Desktop cannot move disk images to a different drive and we will be soon shipping a fix.  

The workaround described will work but **be aware** that it will create a new data disk, dropping any data that was present in the previous disk, if any. If you want to also migrate data, please make sure to copy `docker_data.vhdx` from the old to the new directory before starting up Docker Desktop again.

I am closing this issue as duplicate. You can follow prorgress in the issue linked above.

**Comment by bristledev** [comment]
Great update, ya'll.  My entire ext4.vhdx is gone and replaced with an empty docker_data.vhdx thanks to this change.  Good thing there wasn't anything important in it.  Bravo.  This one issue is the only reason how I know *why* it's gone.
